JASP-Engine/JASP/R/exploratoryfactoranalysis.R

```diff
@@ -27,379 +28,315 @@
 
 
 ExploratoryFactorAnalysis <- function(dataset = NULL, options, perform = "run",
-                                      callback = function(...) 0, ...) {
-  
-  
+                                      callback = function(...) list(status="ok"), state = NULL, ...) {
   
+
   ## call the common initialization function
-  init <- .initializeEFA(dataset, options, perform)
+init <- .initialize(dataset, options, perform)
   
   
   results <- init[["results"]]
   dataset <- init[["dataset"]]
+
+
+  ## initialize result ## ----
+  results[["title"]] = "Exploratory Factor Analysis"
+  meta <- list(
+    list(name="title", type="title"),
+    list(name="factorLoadings", type = "table"),
+    list(name="factorCorrelations", type = "table"),
+    list(name="goodnessOfFit", type="table"),
+    list(name="fitMeasures", type="table"),
+    list(name="pathDiagram", type="image"),
+    list(name="screePlot", type="image"))
+
+  results[[".meta"]] = meta
+
+  ## initialize state ## ----
+
+  stateKey <- list(
+    analysisResults = c(
+      'rotationMethod',
+      'orthogonalSelector',
+      'obliqueSelector',
+      'variables', 
+      'factorMethod',
+      'eigenValuesBox',
+      'numberOfFactors'),
+    pathDiagram = c(
+      'rotationMethod',
+      'orthogonalSelector',
+      'obliqueSelector',
+      'variables', 
+      'factorMethod',
+      'eigenValuesBox',
+      'numberOfFactors',
+      'highlightSlider', 
+      'highlightText',
+      # 'incl_pathDiagram', 
+      'plotWidthPathDiagram',
+      'plotHeightPathDiagram'),
+    screePlot = c(
+      'rotationMethod',
+      'orthogonalSelector',
+      'obliqueSelector',
+      'variables', 
+      'factorMethod',
+      'eigenValuesBox',
+      'numberOfFactors',
+      # 'incl_screePlot',
+      'plotWidthScreePlot',
+      'plotHeightScreePlot')
+    )
+  keep <- NULL
+
```

Comment:
@hummyhummy These comments can be removed

Reply:
removed